source/common/ssl/ssl_socket.cc

```diff
@@ -62,6 +63,9 @@ Network::IoResult SslSocket::doRead(Buffer::Instance& read_buffer) {
         switch (err) {
         case SSL_ERROR_WANT_READ:
           break;
+        case SSL_ERROR_ZERO_RETURN:
+          last_byte = true;
```

Comment:
same as raw transport socket I think it's worth latching this and asserting on further reads, just for sanity checks.